### PR TITLE
feat: User#flags

### PR DESCRIPTION
**Pull request by Fyko**
**Please describe the changes this PR makes and why it should be merged:**
This PR adds support for a new-feature: public user flags.
Since the `public_flags` value is not included on the initial data sending, the `fetchFlags` function is necessary. Will keep as draft until the relevant DAPI issue is merged. https://github.com/discord/discord-api-docs/pull/1490

Props to https://github.com/Keke27210/homer/commit/1f21820b3f8c7419900d58fded21798f1b3e8b4d for the reminder and code in the Discord Developers server.

**Status**
- [x] Code changes have been tested against the Discord API, or there are no code changes
- [x] I know how to update typings and have done so, or typings don't need updating

**Semantic versioning classification:**
- [x] This PR changes the library's interface (methods or parameters added)
  - [ ] This PR includes breaking changes (methods removed or renamed, parameters moved or removed)
- [ ] This PR **only** includes non-code changes, like changes to documentation, README, etc.

### CODE REVIEW

**Comment by Monbrey** [comment]
I don't think I agree with the `UserFlags#fetched` implementatio - I'm not sure it needs to exist at all.

I think `User#flags` should just be a nullable property if they haven't been fetched.
At the very least, `UserFlasg#fetched` only needs to be a getter and not a manually set property.

**Comment by ItsPapaia** [comment]
The return value of `fetchFlags` should be a consistent promise even if the flags are fetched, which doesn't happen right now, as you're just returning the flags if they exist, you should return `Promise.resolve(this.flags)` or make the method async.

File: src/structures/User.js [hunk 1]
```diff
@@ -255,6 +262,22 @@ class User extends Base {
     return equal;
   }
 
+  /**
+   * Fetches this user's flags.
+   * @returns {Promise<UserFlags>}
+   */
+  fetchFlags() {
```
Comment (ItsPapaia) [inline src/structures/User.js:269]: If the flags are fetched, you don't return a promise.
```suggestion
  async fetchFlags() {
```

**Comment by ItsPapaia** [comment]
Also, you don't export them?

File: src/util/UserFlags.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  DISCORD_PARTNER: 1 << 1,
+  HYPESQUAD_EVENTS: 1 << 2,
+  BUGHUNTER_LEVEL1: 1 << 3,
+  HOUSE_BRAVERY: 1 << 6,
+  HOUSE_BRILLANCE: 1 << 7,
+  HOUSE_BALANCE: 1 << 8,
+  EARLY_SUPPORTER: 1 << 9,
+  TEAM_USER: 1 << 10,
+  SYSTEM: 1 << 12,
+  BUGHUNTER_LEVEL2: 1 << 14,
```
Comment (vladfrangu) [inline src/util/UserFlags.js:54]:
```suggestion
  BUGHUNTER_LEVEL2: 1 << 14,
  VERIFIED_BOT: 1 << 16,
  VERIFIED_DEVELOPER: 1 << 17,
```

File: typings/index.d.ts [hunk 3]
```diff
@@ -2980,6 +2986,19 @@ declare module 'discord.js' {
...
+    | 'DISCORD_PARTNER'
+    | 'HYPESQUAD_EVENTS'
+    | 'BUGHUNTER_LEVEL1'
+    | 'HOUSE_BRAVERY'
+    | 'HOUSE_BRILLANCE'
+    | 'HOUSE_BALANCE'
+    | 'EARLY_SUPPORTER'
+    | 'TEAM_USER'
+    | 'SYSTEM'
+    | 'BUGHUNTER_LEVEL2';
```
Comment (vladfrangu) [inline typings/index.d.ts:3000]:
```suggestion
    | 'BUGHUNTER_LEVEL2'
    | 'VERIFIED_BOT'
    | 'VERIFIED_DEVELOPER';
```

File: src/util/UserFlags.js [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+ * @type {Object}
+ * @see {@link https://discordapp.com/developers/docs/resources/user#user-object-user-flags}
+ */
+UserFlags.FLAGS = {
+  DISCORD_EMPLOYEE: 1 << 0,
+  DISCORD_PARTNER: 1 << 1,
+  HYPESQUAD_EVENTS: 1 << 2,
+  BUGHUNTER_LEVEL1: 1 << 3,
+  HOUSE_BRAVERY: 1 << 6,
+  HOUSE_BRILLANCE: 1 << 7,
```
Comment (advaith1) [inline src/util/UserFlags.js:41]:
```suggestion
  HOUSE_BRILLIANCE : 1 << 7,
```
typo, needs to be fixed in 3 places

File: src/util/UserFlags.js [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+ * * `VERIFIED_BOT`
+ * * `VERIFIED_DEVELOPER`
+ * @type {Object}
+ * @see {@link https://discordapp.com/developers/docs/resources/user#user-object-user-flags}
+ */
+UserFlags.FLAGS = {
+  DISCORD_EMPLOYEE: 1 << 0,
+  DISCORD_PARTNER: 1 << 1,
+  HYPESQUAD_EVENTS: 1 << 2,
+  BUGHUNTER_LEVEL1: 1 << 3,
```
Comment (ThaTiemsz) [inline src/util/UserFlags.js:39]:
```suggestion
  BUG_HUNTER_LEVEL_1: 1 << 3,
```
Minor nitpick but could the bug hunter flags use the above letter casing?

**Comment by ItsPapaia** [comment]
The level flags in the typings still use the 'old' styling.

**Comment by advaith1** [comment]
This is now documented (https://github.com/discord/discord-api-docs/pull/1490 was merged) so can this be merged now?

**Pull request merged**